drivers/timer/Kconfig.riscv_machine

```diff
@@ -11,3 +11,14 @@ config RISCV_MACHINE_TIMER
 	help
 	  This module implements a kernel device driver for the generic RISCV machine
 	  timer driver. It provides the standard "system clock driver" interfaces.
+
+config RISCV_MACHINE_TIMER_SYSTEM_CLOCK_DIVIDER
+	int "RISCV Machine Timer system clock divider"
```

Comment:
The value of clk-divider defined in nuclei,machine-timer is used as RISCV_MACHINE_TIMER_SYSTEM_CLOCK_DIVIDER.

I defined the clk-divider on nuclei,machine-timer.yaml because existing boards use machine-timer without devicetree definitions.
I think it strange that  Kconfig.riscv_machine reference to Nuclei-specific device-tree.

I will correcting it to move nuclei,machine-timer.yaml to riscv,machine-timer.yaml, and add description to Kconfig.riscv_machine.

Reply:
Thanks for the explanation.

My understand of this is that this value primarily should be controlled through devicetree.

In that case, I would suggest to remove the prompt here, but keep the default to `0` as that seems to be a sane default.

And then in the `help` section describe that machine timer clock divider should preferable be set in devicetree.
Having this promptless will still allow a board (or  sample if need be) to set a different default value in the board / sample specific Kconfig for those situation where the clock divider has not been specified in devicetree, while still give a strong hint that clock divider preferable should come from devicetree (and not be adjusted for example in a conf file)

Having the default value based on the specific devicetree node placed in SoC specific Kconfig like here 3786ed689b2a41dd1d897355fb7b29aa687b8b67 is good choice.

Small side-note, please let the comment auther resolve the conversation as you don't know if the author of the initial comment agrees to the solution or if there will be follow up questions / comments :wink:

Reply:
I'll make changes follows...
- Removing prompt (Remove string literal after "int" declaration)
- Keep default value to 0
- Add comments to notice that the value should define in devicetree

> Small side-note, please let the comment auther resolve the conversation as you don't know if the author of the initial comment agrees to the solution or if there will be follow up questions / comments 😉

I'm sorry. 
I wasn't Getting used to the review tools. I will take care.

Reply:
> I'll make changes follows...

Great, thanks.

> I wasn't Getting used to the review tools. I will take care. ok_hand

No problem.
You're not the first and won't be the last.

It takes time to get used to good workflows.